pkg/snapshot/restorer/restorer_test.go

```diff
@@ -71,116 +79,522 @@ var _ = Describe("Running Restorer", func() {
+				err = corruptEtcdDir()
+				Expect(err).ShouldNot(HaveOccurred())
+				logger.Infoln("corrupted the etcd dir")
+
+				store, err = snapstore.GetSnapstore(&snapstore.Config{Container: snapstoreDir, Provider: "Local"})
+				Expect(err).ShouldNot(HaveOccurred())
+				baseSnapshot, deltaSnapList, err = miscellaneous.GetLatestFullSnapshotAndDeltaSnapList(store)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				rstr = NewRestorer(store, logger)
+
+				RestoreOptions := RestoreOptions{
+					ClusterURLs:            clusterUrlsMap,
+					ClusterToken:           restoreClusterToken,
+					RestoreDataDir:         restoreDataDir,
+					PeerURLs:               peerUrls,
+					SkipHashCheck:          skipHashCheck,
+					Name:                   restoreName,
+					MaxFetchers:            maxFetchers,
+					EmbeddedEtcdQuotaBytes: embeddedEtcdQuotaBytes,
+					BaseSnapshot:           *baseSnapshot,
+					DeltaSnapList:          deltaSnapList,
+				}
+				logger.Infoln("starting restore while snapshotter is running")
+				err = rstr.Restore(RestoreOptions)
+				Expect(err).Should(HaveOccurred())
+
+			})
+		})
+
+		Context("with etcd data dir not cleaned up before restore", func() {
+			fmt.Println("Testing restore on an existing etcd data directory")
+			It("Should fail to restore", func() {
+
+				deltaSnapshotPeriod := 1
+				wg.Add(1)
+				go populateEtcd(wg, logger, endpoints, errCh, populatorStopCh)
+
+				go stopLoaderAndSnapshotter(wg, 10, 15, populatorStopCh, ssrStopCh)
+
+				logger.Infoln("Starting snapshotter while loading is happening")
+				err = runSnapshotter(logger, deltaSnapshotPeriod, endpoints, ssrStopCh)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				etcd.Server.Stop()
+				etcd.Close()
+
+				//time.Sleep(time.Duration(50 * time.Second))
+				// err = corruptEtcdDir()
+				// Expect(err).ShouldNot(HaveOccurred())
+				// logger.Infoln("corrupted the etcd dir")
+
+				store, err = snapstore.GetSnapstore(&snapstore.Config{Container: snapstoreDir, Provider: "Local"})
+				Expect(err).ShouldNot(HaveOccurred())
+				baseSnapshot, deltaSnapList, err = miscellaneous.GetLatestFullSnapshotAndDeltaSnapList(store)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				rstr = NewRestorer(store, logger)
+
+				RestoreOptions := RestoreOptions{
+					ClusterURLs:            clusterUrlsMap,
+					ClusterToken:           restoreClusterToken,
+					RestoreDataDir:         restoreDataDir,
+					PeerURLs:               peerUrls,
+					SkipHashCheck:          skipHashCheck,
+					Name:                   restoreName,
+					MaxFetchers:            maxFetchers,
+					EmbeddedEtcdQuotaBytes: embeddedEtcdQuotaBytes,
+					BaseSnapshot:           *baseSnapshot,
+					DeltaSnapList:          deltaSnapList,
+				}
+				logger.Infoln("starting restore restore directory exists already")
+				err = rstr.Restore(RestoreOptions)
+				Expect(err).Should(HaveOccurred())
+
+			})
+		})
+
+		Context("while snapshooter is running ", func() {
```

Comment:
Given that this test falls under the scope of server command rather than unit tests for restoration, because the server takes care of the coordination between snapshotter and restorer, I would advise moving this test to a different scope, maybe integration tests.

Reply:
the test is now marked with 'X' and is excluded during the run. I've kept it for reference for refactoring at a later stage once agreed and aligned for handling such scenarios.